sky/utils/command_runner.py

```diff
@@ -138,14 +142,30 @@ def __init__(
             ssh_proxy_command: Optional, the value to pass to '-o
                 ProxyCommand'. Useful for communicating with clusters without
                 public IPs using a "jump server".
+            docker_user: The docker user to use for ssh. If specified, the
+                command will be run inside a docker container which have a ssh
+                server running at port docker_utils.DEFAULT_DOCKER_PORT.
         """
-        self.ip = ip
-        self.ssh_user = ssh_user
         self.ssh_private_key = ssh_private_key
         self.ssh_control_name = (
             None if ssh_control_name is None else hashlib.md5(
                 ssh_control_name.encode()).hexdigest()[:_HASH_MAX_LENGTH])
-        self._ssh_proxy_command = ssh_proxy_command
+        if docker_user:
+            self.ip = 'localhost'
+            self.ssh_user = docker_user
+            if ssh_proxy_command is not None:
+                raise ValueError(
+                    'ssh_proxy_command is not supported if docker is used')
+            self._ssh_proxy_command = lambda ssh: ' '.join(
+                ssh + ssh_options_list(ssh_private_key, None
+                                      ) + ['-W', '%h:%p', f'{ssh_user}@{ip}'])
```

Comment:
It seems possible to do multi-hop jump with `ssh`, with something like the following:
```
ssh -o ProxyCommand="ssh -W %h:%p user1@jump1" -o ProxyCommand="ssh -W %h:%p user2@jump2" user3@target
```

Is it possible for us to support it? We can also do it in another PR. : )